src/controllers/dataset.ts

```diff
@@ -336,3 +340,80 @@ export const getFactTableDefinition = async (req: Request, res: Response) => {
   res.status(200);
   res.json(factTableDto);
 };
+
+async function addDirectoryToZip(
+  zip: JSZip,
+  datasetFiles: Map<string, FileImportDto>,
+  directory: string,
+  fileService: StorageService
+) {
+  const directoryList = await fileService.listFiles(directory);
+  for (const fileEntry of directoryList) {
+    let filename: string;
+    if ((fileEntry as DataLakeFileEntry).name) {
+      const entry = fileEntry as DataLakeFileEntry;
+      if (entry.isDirectory) {
+        await addDirectoryToZip(zip, datasetFiles, `${directory}/${entry.name}`, fileService);
+        continue;
+      }
+      filename = (fileEntry as DataLakeFileEntry).name;
+    } else {
+      filename = fileEntry as string;
+    }
+    const originalFilename = datasetFiles.get(filename)?.filename || filename;
+    zip.file(originalFilename, await fileService.loadBuffer(filename, directory));
+  }
+}
+
+function collectFiles(dataset: Dataset): Map<string, FileImportDto> {
+  const files: Map<string, FileImportDto> = new Map<string, FileImportDto>();
```

Comment:
More personal preference than anything, but generally you don't need to type hint stuff when it's implicitly typed by the assignment, feels a bit redundant in cases like this.

Reply:
Removed the type hint from this line... My past as a Java Dev came through 😆